public/app/features/logs/components/logParser.ts

```diff
@@ -35,20 +35,30 @@ export const getAllFields = (
 export const createLogLineLinks = (hiddenFieldsWithLinks: FieldDef[]): FieldDef[] => {
   let fieldsWithLinksFromVariableMap: FieldDef[] = [];
   hiddenFieldsWithLinks.forEach((linkField) => {
-    linkField.links?.forEach((link: ExploreFieldLinkModel) => {
-      if (link.variables) {
-        const variableKeys = link.variables.map((variable) => {
-          const varName = variable.variableName;
-          const fieldPath = variable.fieldPath ? `.${variable.fieldPath}` : '';
-          return `${varName}${fieldPath}`;
-        });
-        const variableValues = link.variables.map((variable) => (variable.found ? variable.value : ''));
-        fieldsWithLinksFromVariableMap.push({
-          keys: variableKeys,
-          values: variableValues,
-          links: [link],
-          fieldIndex: linkField.fieldIndex,
-        });
+    linkField.links?.forEach((link: LinkModel | ExploreFieldLinkModel) => {
+      if ('variables' in link) {
+        if (link.variables.length > 0) {
+          // convert ExploreFieldLinkModel to LinkModel by omitting variables field
+          const fieldDefFromLink: LinkModel = {
```

Comment:
Not really understanding why this is needed.
```
export interface ExploreFieldLinkModel extends LinkModel<Field> {
  variables: VariableInterpolation[];
}
```
so we should be able to pass `ExploreFieldLinkModel` wherever a `LinkModel` is required. Also the `FieldDef` has ` links?: Array<LinkModel<Field>> | ExploreFieldLinkModel[];` so that should not create problems either.